integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -0,0 +1,633 @@
+        groupName = GROUP_NAME;
+      }
+      attributeOptionsBuilder.setGroup(groupName);
+    }
+    if (uid != DEFAULT_UID) {
+      String userName = AlluxioFuseUtils.getUserName(uid);
+      if (userName.isEmpty()) {
+        // This should never be reached since input uid is always valid
+        LOG.error("Failed to get user name from uid {}, fallback to {}", uid, USER_NAME);
+        userName = USER_NAME;
+      }
+      attributeOptionsBuilder.setOwner(userName);
+    }
+    SetAttributePOptions setAttributePOptions =  attributeOptionsBuilder.build();
+    if (gid != DEFAULT_GID || uid != DEFAULT_UID) {
+      LOG.debug("Set attributes of path {} to {}", uri, setAttributePOptions);
+      mFileSystem.setAttribute(uri, setAttributePOptions);
+    }
+  }
+
+  @Override
+  public int create(String path, long mode, FuseFileInfo fi) {
+    return AlluxioFuseUtils.call(LOG, () -> createInternal(path, mode, fi),
+        "create", "path=%s,mode=%o", path, mode);
+  }
+
+  private int createInternal(String path, long mode, FuseFileInfo fi) {
+    final AlluxioURI uri = mPathResolverCache.getUnchecked(path);
+    if (uri.getName().length() > MAX_NAME_LENGTH) {
+      LOG.error("Failed to create {}: file name longer than {} characters",
+          path, MAX_NAME_LENGTH);
+      return -ErrorCodes.ENAMETOOLONG();
+    }
+    try {
+      FileOutStream os = mFileSystem.createFile(uri,
+          CreateFilePOptions.newBuilder()
+              .setMode(new Mode((short) mode).toProto())
+              .build());
+      long fid = mNextOpenFileId.getAndIncrement();
+      mCreateFileEntries.put(fid, os);
+      fi.fh.set(fid);
+      setUserGroupIfNeeded(uri);
+    } catch (Throwable e) {
+      LOG.error("Failed to getattr {}: ", path, e);
+      return -ErrorCodes.EIO();
+    }
+    return 0;
+  }
+
+  @Override
+  public int getattr(String path, FileStat stat) {
+    return AlluxioFuseUtils.call(
+        LOG, () -> getattrInternal(path, stat), "getattr", "path=%s", path);
+  }
+
+  private int getattrInternal(String path, FileStat stat) {
+    final AlluxioURI uri = mPathResolverCache.getUnchecked(path);
+    try {
+      URIStatus status = mFileSystem.getStatus(uri);
+      long size = status.getLength();
+      stat.st_size.set(size);
+
+      // Sets block number to fulfill du command needs
+      // `st_blksize` is ignored in `getattr` according to
+      // https://github.com/libfuse/libfuse/blob/d4a7ba44b022e3b63fc215374d87ed9e930d9974/include/fuse.h#L302
+      // According to http://man7.org/linux/man-pages/man2/stat.2.html,
+      // `st_blocks` is the number of 512B blocks allocated
+      stat.st_blocks.set((int) Math.ceil((double) size / 512));
+
+      final long ctime_sec = status.getLastModificationTimeMs() / 1000;
+      // Keeps only the "residual" nanoseconds not caputred in citme_sec
+      final long ctime_nsec = (status.getLastModificationTimeMs() % 1000) * 1000;
```

Comment:
Is this supposed to be nanoseconds? Aren't there `1000000ns` in `1ms`?

Reply:
good catch, fixed